cumulus/tasks/sf-sns-report/index.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const get = require('lodash.get');
+const { StepFunction } = require('@cumulus/ingest/aws');
+const { setGranuleStatus, sns } = require('@cumulus/common/aws');
+const errors = require('@cumulus/common/errors');
+const cumulusMessageAdapter = require('@cumulus/cumulus-message-adapter-js');
+
+/**
+ * Determines if there was a valid exception in the input message
+ *
+ * @param {Object} event - aws event object
+ * @returns {boolean} true if there was an exception, false otherwise
+ */
+function eventFailed(event) {
+  // event has exception
+  // and it is needed to avoid flagging cases like "exception: {}" or "exception: 'none'"
+  if (event.exception && (typeof event.exception === 'object') &&
```

Comment:
The first check for event.exception is not necessary.  If it isn't set, then `(typeof event.exception === 'object')` will be false.  Can just be:

```
function eventFailed(event) {
  // event has exception
  // and it is needed to avoid flagging cases like "exception: {}" or "exception: 'none'"
  if ((typeof event.exception === 'object') &&
      (Object.keys(event.exception).length > 0)) return true;

  // Error and error keys are not part of the cumulus message
  // and if they appear in the message something is seriously wrong
  if (event.Error || event.error) return true;

  return false;
}
```

Reply:
This example is also formatted so it's a little easier to read.

Reply:
updated

Reply:
actually ``(event.exception)` check is need to take care of the case `event.exception=null`, otherwise `Object.keys(event.exception)` will throw exception.